### PR TITLE
Correct documentation for parallel= option to astropy.test()

### LINKED ISSUE #7547: Incorrect documentation for parallel= option to astropy.test()
The documentation for `astropy.test()` states that setting setting the option `parallel=-1` should result in automatically using all cores. However, it currently does not accept negative numbers; instead, the desired effect occurs when you set `parallel='auto'`.

## The documentation

```
parallel : int, optional
    When provided, run the tests in parallel on the specified
    number of CPUs.  If parallel is negative, it will use the all
    the cores on the machine.  Requires the ``pytest-xdist`` plugin.
```

## `parallel=-1` fails

```
>>> astropy.test(parallel=-1)
/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)
============================= test session starts ==============================
platform darwin -- Python 3.6.5, pytest-3.5.1, py-1.5.3, pluggy-0.6.0

Running tests with Astropy version 3.1.dev21994.
Running tests in local/lib/python3.6/site-packages/astropy.

Date: 2018-06-07T08:59:57

Platform: Darwin-17.6.0-x86_64-i386-64bit

Executable: /opt/local/bin/python

Full Python Version: 
3.6.5 (default, Mar 29 2018, 15:37:32) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.14.3
Scipy: 1.1.0
Matplotlib: 2.2.2
h5py: 2.7.0
Pandas: 0.23.0
Cython: 0.28.2
astropy_helpers: 2.0.2
Using Astropy options: remote_data: none.

Matplotlib: 2.2.2
Freetype: 2.8.1
rootdir: /Users/lpsinger, inifile:
plugins: celery-4.2.0rc2, remotedata-0.2.0, pycodestyle-1.2.1, postgresql-1.3.2, pep8-1.0.6, openfiles-0.2.0, flakes-2.0.0, flake8-1.0.1, doctestplus-0.1.2, cython-0.1.0, cov-2.5.1, arraydiff-0.1, xdist-1.14, mpl-0.9, hypothesis-3.56.9
ERROR: MISSING test execution (tx) nodes: please specify --tx

4
```

## `parallel='auto'` works

```
>>> astropy.test(parallel='auto')
/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)
============================= test session starts ==============================
platform darwin -- Python 3.6.5, pytest-3.5.1, py-1.5.3, pluggy-0.6.0

Running tests with Astropy version 3.1.dev21994.
Running tests in local/lib/python3.6/site-packages/astropy.

Date: 2018-06-07T09:02:24

Platform: Darwin-17.6.0-x86_64-i386-64bit

Executable: /opt/local/bin/python

Full Python Version: 
3.6.5 (default, Mar 29 2018, 15:37:32) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.14.3
Scipy: 1.1.0
Matplotlib: 2.2.2
h5py: 2.7.0
Pandas: 0.23.0
Cython: 0.28.2
astropy_helpers: 2.0.2
Using Astropy options: remote_data: none.

Matplotlib: 2.2.2
Freetype: 2.8.1
rootdir: /Users/lpsinger, inifile:
plugins: celery-4.2.0rc2, remotedata-0.2.0, pycodestyle-1.2.1, postgresql-1.3.2, pep8-1.0.6, openfiles-0.2.0, flakes-2.0.0, flake8-1.0.1, doctestplus-0.1.2, cython-0.1.0, cov-2.5.1, arraydiff-0.1, xdist-1.14, mpl-0.9, hypothesis-3.56.9
gw0 [10907] / gw1 [10907] / gw2 [10907] / gw3 [10907]
scheduling tests via LoadScheduling
.............etc....
```

I can reproduce this with both astropy 3.0.1 and master (as of 0aecec17bc6a4d401d550f6cf2f80e8014832a58).

### PR DESCRIPTION
**Pull request by lpsinger**
Fixes #7547.

### CODE REVIEW

**Comment by pllim** [comment]
Usually, doc update does not need a change log entry. :smile:

**Comment by lpsinger** [comment]
> Usually, doc update does not need a change log entry.

Thanks, I removed the change log entry.

File: docs/development/testguide.rst [hunk 1]
```diff
@@ -237,8 +237,8 @@ commandline option.  For example, to use 4 processes::
 
     python setup.py test --parallel=4
 
-Pass a negative number to ``'--parallel'`` to create the same number of
-processes as cores on your machine.
+Pass ``'--parallel=auto'`` to create the same number of processes as
```
Comment (pllim) [inline docs/development/testguide.rst]: Shouldn't it be ``--parallel='auto'``? (Note the placement of first quote)

Comment (lpsinger) [reply docs/development/testguide.rst]: Actually, no quote is needed.

**Comment by bsipocz** [comment]
Thanks @lpsinger!

**Pull request merged**